src/storage/data_object_type_registry.rs

```diff
@@ -0,0 +1,154 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use rstd::prelude::*;
+use parity_codec::Codec;
+use parity_codec_derive::{Encode, Decode};
+use srml_support::{StorageMap, StorageValue, decl_module, decl_storage, decl_event, ensure, Parameter};
+use runtime_primitives::traits::{SimpleArithmetic, As, Member, MaybeSerializeDebug, MaybeDebug};
+use system::{self, ensure_root};
+use crate::traits;
+
+pub trait Trait: system::Trait + MaybeDebug
+{
+    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+
+    type DataObjectTypeID: Parameter + Member + SimpleArithmetic + Codec + Default + Copy
+        + As<usize> + As<u64> + MaybeSerializeDebug + PartialEq;
+}
+
+
+static MSG_REQUIRE_NEW_DO_TYPE: &str = "New Data Object Type required; the provided one seems to be in use already!";
+static MSG_DO_TYPE_NOT_FOUND: &str = "Data Object Type with the given ID not found!";
+static MSG_REQUIRE_DO_TYPE_ID: &str = "Can only update Data Object Types that are already registered (with an ID)!";
+
+const DEFAULT_FIRST_DATA_OBJECT_TYPE_ID: u64 = 1;
+
+#[derive(Clone, Encode, Decode, PartialEq)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct DataObjectType<T: Trait>
+{
+    // If the OT is registered, an ID must exist, otherwise it's a new OT.
+    pub id: Option<T::DataObjectTypeID>,
```

Comment:
I would suggest changing `id` type to just `T::DataObjectTypeId` and creating an additional struct just for updates:
```
pub struct DataObjectTypeUpdate<T: Trait> {
  description: Option<Vec<u8>>,
  active: Option<bool>,
}
```

And in `update_*` function, update only those fields of stored value that are `Some` in `DataObjectTypeUpdate`

Reply:
Ok, why? I suppose it may keep transaction sizes down. I suppose that could be worth it, but is that the rationale?

Reply:
The tx size is certainly a benefit, but I think the rationale behind alex's suggestion is so that you wouldn't have to change the function signature in the future if you introduce new fields to theDataObjectType type. (I don't want to put words in his mouth but that is how I understood it when he recommended the same for the members module [update_profile()](https://github.com/Joystream/substrate-runtime-joystream/blob/b5926390b1bf9e94fb60cc1c17a615a14a8fb7e2/src/membership/members.rs#L241) method)

Reply:
I actually didn't look at the function signature (i assumed there was one parameter per field).. so my point doesn't actually make sense.

Reply:
Because object's `id` is used quite often (at least on UI: list stored objects, view details of stored objects, etc.) and having it as an Option will require additional check `storedObj.id.isSome ? ...then : else...` everywhere. Comparing to update – this operation is rate and exists only on Update Form.